pkg/virtctl/root.go

```diff
@@ -133,8 +134,36 @@ func Execute() {
 	log.InitializeLogging(programName)
 	cmd, clientConfig := NewVirtctlCommand()
 	if err := cmd.Execute(); err != nil {
-		version.CheckClientServerVersion(&clientConfig, cmd)
-		fmt.Fprintln(cmd.Root().ErrOrStderr(), strings.TrimSpace(err.Error()))
+		if versionErr := CheckClientServerVersion(clientConfig); versionErr != nil {
+			cmd.PrintErrln(versionErr)
+		}
+		cmd.PrintErrln(err)
 		os.Exit(1)
 	}
 }
+
+func CheckClientServerVersion(clientConfig clientcmd.ClientConfig) error {
```

Comment:
Do we need to export it ?

Reply:
No, AFAICT it's just to allow for more targeted tests.

I don't think we have any coding conventions for this but I'd happily see it documented as an anti-pattern with folks forced to assert the true public path (`Execute` in this case) instead.

Reply:
I agree, let me rework this.